### PR TITLE
Fix sql with strange behaviour, locatealong on startpoint or endpoint

### PR DESCRIPTION
**Pull request by LePetitTim**

### CODE REVIEW

File: geotrek/core/sql/20_evenements.sql [hunk 1]
```diff
@@ -102,13 +103,24 @@ BEGIN
...
         SELECT geom, decallage INTO egeom, t_offset FROM e_t_evenement e WHERE e.id = eid;
-
         -- RAISE NOTICE '% % % %', (t_offset = 0), (egeom IS NULL), (ST_IsEmpty(egeom)), (ST_X(egeom) = 0 AND ST_Y(egeom) = 0);
         IF t_offset = 0 OR egeom IS NULL OR ST_IsEmpty(egeom) OR (ST_X(egeom) = 0 AND ST_Y(egeom) = 0) THEN
-            SELECT ST_GeometryN(ST_LocateAlong(ST_AddMeasure(ST_Force2D(t.geom), 0, 1), et.pk_debut, e.decallage), 1)
-                INTO egeom
+            -- ST_LocateAlong can give no point when we try to get the startpoint or the endpoint of the line
+            SELECT et.pk_debut INTO position_point FROM e_r_evenement_troncon et WHERE et.evenement = eid;
+            IF (position_point = 0) THEN
+                SELECT ST_GeometryN(ST_StartPoint(t.geom), 1) INTO egeom
```
Comment (gutard) [inline geotrek/core/sql/20_evenements.sql]: ST_GeometryN() semble inutile ici

File: geotrek/core/sql/20_evenements.sql [hunk 2]
```diff
@@ -102,13 +103,24 @@ BEGIN
...
+                FROM e_t_evenement e, e_r_evenement_troncon et, l_t_troncon t
+                WHERE e.id = eid AND et.evenement = e.id AND et.troncon = t.id;
+            ELSIF (position_point = 1) THEN
+                SELECT ST_GeometryN(ST_EndPoint(t.geom), 1) INTO egeom
```
Comment (gutard) [inline geotrek/core/sql/20_evenements.sql]: ST_GeometryN() semble inutile ici

Comment (LePetitTim) [reply geotrek/core/sql/20_evenements.sql]: Done

File: geotrek/core/sql/20_evenements.sql (hunk 1 above)
Comment (LePetitTim) [reply geotrek/core/sql/20_evenements.sql]: Done

**Pull request merged**